include/swift/AST/Module.h

```diff
@@ -209,7 +208,23 @@ class ModuleDecl : public DeclContext, public TypeDecl {
 
 public:
   typedef ArrayRef<Located<Identifier>> AccessPathTy;
-  typedef std::pair<ModuleDecl::AccessPathTy, ModuleDecl*> ImportedModule;
+  struct ImportedModule {
+    ModuleDecl::AccessPathTy accessPath;
+    ModuleDecl *importedModule;
+  
+    /// Will not accept a null moduleDecl pointer.
+    ImportedModule(ModuleDecl::AccessPathTy accessPath,
```

Comment:
It would be nice to have a short doc comment for the struct and fields. Say something like:

   ```
   /// Convenience struct to keep track of a module along with its access path.
   struct ImportedModule {
     /// The access path from an import: `import Foo.Bar` -> `Foo.Bar`.
     ModuleDecl::AccessPathTy accessPath;

     /// The actual module corresponding to the import.
     ///
     /// Invariant: The pointer is non-null.
     ModuleDecl *importedModule;
   }
   ```

   Thoughts/suggestions for improvement? I think it would be better to document the non-null pointer with the field than at the constructor because it is an invariant of the type, not just of that one particular constructor.

Reply:
I agree. What you have in the example is about the right amount of information. One thing--I have been thinking of this as access path + module, because that's the order of the original pair, but you wrote 'module along with its access path' above, and it might make more sense to put the module pointer first and access path second. I'm not sure about that though. What do you think?

Reply:
I think having the module pointer first is ok too. 👍 I wrote "module along with its access path" because that seemed to flow naturally.

Reply:
I put the doc comment in (as you wrote it) in 1d625d8. Reordering the fields needs some work to clear some warnings.

Reply:
Yeah, in the constructor, the initialization order needs to match the layout order, I think? However, I think you shouldn't need to swap the order of the arguments to the constructor, that might be a lot of work (because you'll need to swap arguments at all the use sites of the constructor), for what seems like little benefit.

Reply:
Just double-checking; since you brought up some warnings here, were those fixed in one of the later commits/are you planning to address it later?

Reply:
I switched it back to the original ordering with accessPath first, importedModule second in the struct and in the constructors. It seems to be OK. I don't think there's any benefit other than the semantic improvement. I could put in a TODO for it, or just hunt down all the uses, if you think it is worth it. I have already hunted down all the uses once or twice in the last little bit so I could find them again.

Reply:
It's fine, I don't think it's a big deal. 👍